js/ui/status/brightness.js

```diff
@@ -56,11 +62,34 @@ const Indicator = new Lang.Class({
 
     },
 
-    _sliderChanged: function(slider, value) {
-        let percent = value * 100;
+    _updateBrightness: function() {
+        let percent = this._sliderValue * 100;
         this._proxy.Brightness = percent;
     },
 
+    _sliderChanged: function(slider, value) {
+        this._sliderValue = value;
+
+        // It's ok to change the brightness here only if the slider didn't
+        // change because of a mouse dragging event (e.g. keyboard), otherwise
+        // it could push changes faster than how they can actually be handled.
+        if (!this._sliderIsDragging)
+            this._updateBrightness();
+    },
+
+    _sliderDragBegan: function(slider) {
+        this._sliderIsDragging = true;
+    },
+
+    _sliderDragEnded: function(slider) {
+        // We only need to update the brightness here for mouse dragging events,
+        // since other events will update it on the 'value-changed' callback.
+        if (this._sliderIsDragging)
```

Comment:
Is this really needed here? Shouldn't we assume that the slider must be dragging if you have a _drag-end_ event? Or does it receive a drag event when the value changes?
Also without this _if_ I don't think the comment is very valuable here, it's clear that this is for a drag event related change.

Reply:
You're perfectly right, this is a mistake, not sure why I added it back. I'll remove it